pkg/agent/openflow/pipeline.go

```diff
@@ -41,38 +41,127 @@ import (
+	//   defined in file pkg/ovs/openflow/interfaces.go. If you want to add a new stage, please discuss with maintainers
+	//   or developers of Antrea.
+	// - Choose a target pipeline. All pipelines are defined in file pkg/ovs/openflow/interfaces.go. If you want to add a
+	//   new pipeline, please discuss with maintainers or developers of Antrea.
+	// - The most important thing is to choose the right place to declare the new table. For example:
+	//     * If you want to add a table called `DummyTable` between `SpoofGuardTable` and `IPv6Table` in pipeline for IP,
+	//       then the table should be declared as follows:
+	//       ```go
+	//          SpoofGuardTable  = newTable("SpoofGuard", binding.ValidationStage, binding.PipelineIP)
+	//          DummyTable       = newTable("Dummy", binding.ValidationStage, binding.PipelineIP)
+	//	        IPv6Table        = newTable("IPv6", binding.ValidationStage, binding.PipelineIP)
+	//	        IPClassifierTable = newTable("IPClassifier", binding.ValidationStage, binding.PipelineIP)
+	//       ```
+	//      * If you want to add a table called `DummyTable` just before `ARPResponderTable` in pipeline for ARP, then the
+	//        table should be declared as follows:
+	//       ```go
+	//          DummyTable        = newTable("Dummy", binding.OutputStage, binding.PipelineARP)
+	//          ARPResponderTable = newTable("ARPResponder", binding.OutputStage, binding.PipelineARP)
+	//       ```
+	//       * If you want to add a table called `DummyTable` just after `ConntrackStateTable` in pipeline for ARP, then
+	//         the table should be declared as follows:
+	//       ```go
+	//          SNATConntrackTable  = newTable("SNATConntrackZone", binding.ConntrackStateStage, binding.PipelineIP)
+	//          ConntrackTable      = newTable("ConntrackZone", binding.ConntrackStateStage, binding.PipelineIP)
+	//          ConntrackStateTable = newTable("ConntrackState", binding.ConntrackStateStage, binding.PipelineIP)
+	//	        DummyTable          = newTable("Dummy", binding.ConntrackStateStage, binding.PipelineIP)
+	//       ```
+	//  - Reference the newly added table in a feature in file pkg/agent/openflow/feature.go. The table can be referenced by
+	//    multiple features if multiple features need to install flows in the table. Note that, if the newly added table
+	//    is not referenced by any feature or the features referencing the table are all inactivated, then the table will
+	//    not be realized in OVS; if the newly added table is referenced by one or more features and at least on feature
+	//    is activated, then the table will be realized at desired position in OVS pipeline.
+	//  - For pipeline modularity, action GotoTable is no longer recommended. For packet forwarding within a stage, action
+	//    ResubmitToTables is recommended; for packet forwarding between stages, action GotoStage is recommended.
```

Comment:
@jianjuns @tnqn Could you also help review the newly added comments about how to add a new table? Thanks

Reply:
@jianjuns, From line 48 to line 95 in this file.